integration-cli/docker_cli_build_test.go

```diff
@@ -4004,30 +3992,15 @@ RUN    [ "$abc" = '"foo"' ]
 
 ENV    abc=ABC
 RUN    [ "$abc" = "ABC" ]
-ENV    def=${abc:-DEF}
-RUN    [ "$def" = "ABC" ]
-ENV    def=${ccc:-DEF}
-RUN    [ "$def" = "DEF" ]
-ENV    def=${ccc:-${def}xx}
-RUN    [ "$def" = "DEFxx" ]
-ENV    def=${def:+ALT}
-RUN    [ "$def" = "ALT" ]
-ENV    def=${def:+${abc}:}
-RUN    [ "$def" = "ABC:" ]
-ENV    def=${ccc:-\$abc:}
-RUN    [ "$def" = '$abc:' ]
-ENV    def=${ccc:-\${abc}:}
-RUN    [ "$def" = '${abc:}' ]
-ENV    mypath=${mypath:+$mypath:}/home
-RUN    [ "$mypath" = '/home' ]
-ENV    mypath=${mypath:+$mypath:}/away
-RUN    [ "$mypath" = '/home:/away' ]
+ENV    def1=${abc:-DEF} def2=${ccc:-DEF}
+ENV    def3=${ccc:-${def2}xx} def4=${abc:+ALT} def5=${abc:+${abc}:} def6=${ccc:-\$abc:} def7=${ccc:-\${abc}:}
```

Comment:
Double check me on this one but I think `def5=${abc:+${abc}:}` needs to use something other than "abc" in the 2nd spot otherwise we won't know whether it used the first or section variable. I think `def5=${abc:+${def2}:}` should do it - but then you need to change $def5's result in the next line to "DEF" not "ABC".  

Also, the previous version of that test assigned the result to the same var being used in the expression - I think we've lost that aspect of the test now.